src/pages/reservationsPage.ts

```diff
@@ -0,0 +1,268 @@
+        case ScheduleWeekDays.W:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.Th:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.F:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.Sa:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+      }
+    } 
+  }
+
+  async saveModalChanges(){
+    await this.saveModalChangesButton.click()
+  }
+
+  async clickNewlyCreatedSchedule(schedulName: string): Promise<void>{
+    const elements = await this.allScheduleTitles.all();
+    if(await this.allScheduleTitles.count() !== 0){
+
+      for(const element of elements){
+        if(await element.textContent() === schedulName){
+          await element.click()
+          return;
+        }
+      }
+    }else {
+      throw new Error("schedule not found");
+    }
+  } 
+
+  async openOverideSection(): Promise<void> {
+    this.newOverrideButton.click()
+  }
+
+  async insertOverrideName({overrideName, amount}: rateOverrideData): Promise<void> {
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    const elements = await this.page.locator('.schedules-modal-section-container .gap-7').all()
+  let overrideNameInputValue 
+  for(const override of elements){
+    overrideNameInputValue = await override.locator(".name-rate-group input.input").inputValue()
+        if(overrideNameInputValue === ""){
+          await override.locator(".name-rate-group input.input").fill(overrideName)
+        }
+  }
+    if(await this.rateOverrideContainerList.count() === 0){
+      console.log('aici crapa la linia 208')
+      this.overrideNameInput.fill(overrideName);
+    }
+  }
+
+  async selectOverrideRate({overrideName, amount}: rateOverrideData): Promise<void> {
+    const rateOverrideElements = await this.page.locator('.schedules-modal-section-container .gap-7').all() 
+    let overrideNameInputValue    
+      for(const override of rateOverrideElements){
+        overrideNameInputValue = await override.locator(".name-rate-group input.input").inputValue()
+        if(overrideNameInputValue === overrideName || overrideNameInputValue === ""){
+          await override.locator('.name-rate-group [aria-autocomplete="list"]').click()
+          await override.locator('.name-rate-group [aria-autocomplete="list"]').press("Enter")
+        }
+      }
+    if(await this.rateOverrideContainerList.count() === 0){
+      this.overideSelectInput.click();
+      this.overideSelectInput.press("Enter");
+    }
+  }
+
+  async insertOverrideAmount({overrideName, amount}: rateOverrideData): Promise<void> {
+    const rateOverrideElements = await this.page.locator('.schedules-modal-section-container .gap-7').all() 
```

Comment:
please use page objects for all hardcoded locators in steps

Reply:
done. I used hardcoded locators just when i need to extend an already defined locator.